lib/src/http/common/request.dart

```diff
@@ -172,6 +172,15 @@ abstract class CommonRequest extends Object
         'The content-length of a request cannot be set manually when the request body is known in advance.');
   }
 
+  Duration get retryTimeoutThreshold {
+    var threshold = timeoutThreshold;
+    if (autoRetry.increaseTimeoutOnRetry) {
+      threshold = timeoutThreshold * autoRetry.numAttempts;
+    }
+
+    return threshold;
+  }
```

Comment:
Since we're not using this in tests, I'd vote for making this private just to avoid confusion on the public API between `timeoutThreshold` and this.

Reply:
Oh, right, this is Dart.